src/ar/hentaislayer/src/eu/kanade/tachiyomi/extension/ar/hentaislayer/HentaiSlayer.kt

```diff
@@ -0,0 +1,245 @@
+    override fun popularMangaRequest(page: Int) = GET("$baseUrl/manga?page=$page", headers)
+
+    override fun popularMangaSelector() = "div > div:has(div#card-real)"
+
+    override fun popularMangaFromElement(element: Element) = SManga.create().apply {
+        element.selectFirst("div#card-real a")?.run {
+            setUrlWithoutDomain(absUrl("href"))
+            selectFirst("figure")?.run {
+                selectFirst("img.object-cover")?.run {
+                    thumbnail_url = imgAttr()
+                    title = attr("alt")
+                }
+                genre = select("span p.drop-shadow-sm").text()
+            }
+        }
+        genre = element.select("span a[href*='?genre=']")
+            .map { it.text() }
+            .let { listOf(genre) + it }
+            .joinToString()
+    }
+
+    override fun popularMangaNextPageSelector() = "ul.pagination > li:last-child:not(.pagination-disabled)"
+
+    // =============================== Latest ===============================
+    override fun latestUpdatesRequest(page: Int) = GET("$baseUrl/latest-${getLatestTypes()}?page=$page", headers)
+
+    override fun latestUpdatesSelector() = popularMangaSelector()
+
+    override fun latestUpdatesFromElement(element: Element) = popularMangaFromElement(element)
+
+    override fun latestUpdatesNextPageSelector() = popularMangaNextPageSelector()
+
+    // =============================== Search ===============================
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = "$baseUrl/manga?title=$query".toHttpUrl().newBuilder()
+        filters.forEach { filter ->
+            when (filter) {
+                is TypeFilter -> url.addQueryParameter("type", filter.toUriPart())
+                is StatusFilter -> url.addQueryParameter("status", filter.toUriPart())
+                is GenresFilter ->
+                    filter.state
+                        .filter { it.state }
+                        .forEach { url.addQueryParameter("genre[]", it.uriPart) }
+                else -> {}
+            }
+        }
+
+        url.addQueryParameter("page", page.toString())
+        return GET(url.build(), headers)
+    }
+
+    override fun searchMangaSelector() = popularMangaSelector()
+
+    override fun searchMangaFromElement(element: Element) = popularMangaFromElement(element)
+
+    override fun searchMangaNextPageSelector() = popularMangaNextPageSelector()
+
+    // =========================== Manga Details ============================
+    override fun mangaDetailsParse(document: Document) = SManga.create().apply {
+        document.selectFirst("main section")?.run {
+            selectFirst("img#manga-cover")?.run {
+                thumbnail_url = imgAttr()
+                title = attr("alt")
+            }
+            selectFirst("section > div:nth-child(1) > div:nth-child(1) > div:nth-child(2) > div:nth-child(2)")?.run {
```

Comment:
is there a better selector available?

Reply:
I don't think there is. This site has bad code structured.